src/tpm2.c

```diff
@@ -5648,6 +5675,67 @@ UINT16 TPM2_GetVendorID(void)
     return vid;
 }
 
+/* Stores nameAlg + the digest of nvPublic in buffer, total size in size */
+int TPM2_HashNvPublic(TPMS_NV_PUBLIC* nvPublic, byte* buffer, UINT16* size)
+{
+    int rc;
+#ifndef WOLFTPM2_NO_WOLFCRYPT
+    int hashSize, nameAlgValue, nameAlgSize;
+    wc_HashAlg hash;
+    enum wc_HashType hashType;
+    byte appending[sizeof(TPMS_NV_PUBLIC)];
+    TPM2B_DIGEST digest;
```

Comment:
This is not guaranteed to be max digest + 2. Use TPM2B_NAME with .name. That is... `TPMT_HA`.

Reply:
Very good point - indeed TPM2B_DIGEST buffer can not fit digest+2, but only digest size.

I will use TPM2B_DATA , because this is the type with buffer of size TPMT_HA
```

typedef struct TPM2B_DATA {
    UINT16 size;
    BYTE buffer[sizeof(TPMT_HA)];
} TPM2B_DATA;

```